RuItUnion.FeedbackBot.Data.Old/Repository.cs

```diff
@@ -1,58 +1,29 @@
 using System.Diagnostics.CodeAnalysis;
 using Microsoft.EntityFrameworkCore;
-using Microsoft.Extensions.Options;
+using Microsoft.Extensions.Logging;
 
-namespace Bot;
+namespace RuItUnion.FeedbackBot.Data.Old;
 
-public interface IRepository
+internal interface IRepository
 {
     void Create<T>(T entity) where T : class, IEntity;
     bool TryRead<T>(long id, [NotNullWhen(true)] out T? entity) where T : class, IEntity;
     bool TryUpdate<T>(T entity) where T : class, IEntity;
     void Delete<T>(long id) where T : class, IEntity;
 }
 
-public class Repository(ILogger<Repository> _logger, IOptions<AppOptions> _options) : IRepository
+public class Repository(ILogger<Repository> logger, string connectionString) : IRepository
 {
-    public class DatabaseContext(IOptions<AppOptions> _options) : DbContext
-    {
-        public DbSet<Topic> Topic { get; set; } = default!;
-        public DbSet<User> Users { get; set; } = default!;
-        public DbSet<Reply> Replies { get; set; } = default!;
-
-        protected override void OnConfiguring(DbContextOptionsBuilder optionsBuilder)
-        {
-            optionsBuilder.UseNpgsql(_options.Value.DbConnectionString);
-        }
-
-        protected override void OnModelCreating(ModelBuilder modelBuilder)
-        {
-            modelBuilder.Entity<Topic>().Property(x => x.Version).IsConcurrencyToken();
-            modelBuilder.Entity<User>().Property(r => r.Version).IsConcurrencyToken();
-            modelBuilder.Entity<Reply>().Property(r => r.Version).IsConcurrencyToken();
-
-            modelBuilder.Entity<Topic>().HasMany(x => x.Replies).WithOne(x => x.Topic);
-            modelBuilder.Entity<Topic>()
-                .HasOne(x => x.User)
-                .WithOne(x => x.Topic)
-                .HasForeignKey<Topic>(x => x.UserId);
-            modelBuilder.Entity<User>()
-                .HasOne(x => x.Topic)
-                .WithOne(x => x.User)
-                .HasForeignKey<User>(x => x.TopicId);
-        }
-    }
-
     public void Create<T>(T entity) where T : class, IEntity

```

Comment:
- Suggest to use `IDbContextFactory` for creating `DbContext`
- Suggest to use `async` methods for db requests

For example:

```csharp
public class Repository(
    ILogger<Repository> logger,
    IDbContextFactory<Repository.DatabaseContext> dbContextFactory
    ) : IRepository

    public async Task CreateAsync<T>(T entity, CancellationToken token) where T : class, IEntity
    {
        await using DatabaseContext db = await dbContextFactory.CreateDbContextAsync(token);
        db.Set<T>().Add(entity);
        await db.SaveChangesAsync(token);
    }

```

Reply:
These methods are not used in the new version, removed them from the code